docs/user-guide/plugins.md

```diff
@@ -246,11 +244,24 @@ event.
     Parameters:
     : __page:__ `mkdocs.nav.Page` instance
     : __config:__ global configuration object
-    : __site_navigation__: global navigation object
+    : __site_navigation:__ global navigation object
 
     Returns:
     : `mkdocs.nav.Page` instance
 
+##### on_load_page_markdown
+
+:   The `load_page_markdown` event can replace the default loading mechanism to
+    load the contents of the file contents.
```

Comment:
I realize you removed some of the text I suggested changing (by adding it to the "return" description -- which makes sense), but we still have "contents of the file contents" here. There are to many "contents" which makes it awkward. Let's change it to "contents of the file from the system". I realize your use case may not actually be reading "from the system" but that makes it clear which default behavior this overrides.

Reply:
thanks again, i think by employing a subjunctive it is described peoperly.

Reply:
But that implies that it should not read from an actual file. As if the loading mechanism **must** fake file reading. Of course, it can do actual file reading.

Let's try this instead:

> The `load_page_markdown` event can replace the default mechanism of reading the contents of a file from the file system.

Here, we are stating what the default is and that it can be replaced. Nowhere do we state what to replace it with. That is left as an exercise for the reader to work out (as it should be). No need to imply artificial limits in the documentation.

Interestingly, by changing the verb to "read" it occurs to me that perhaps a better name for the event would be `on_read_page` (or `on_page_read`) as that is actually what it is doing. "Loading" involves both reading and separating the meta-data. I didn't suggest "load_page" before because that would give the impression of a much larger operation than simply reading the file.

Reply:
the proposal lgtm. in my understanding loading and reading would be synonymous in this context, but you're the native speaker. what's bugging me is, that 'page', imo, points to the `Page` class which is the agent here, but not the object that is read. what about `page_read_markdown`?

Reply:
that's what i have now, notice the amendment to `on_page_markdown`.

```markdown
##### on_page_read_markdown

:   The `page_read_markdown` event can replace the default mechanism of reading 
    the contents of a file from the file system.

    Parameters:
    : __page:__ `mkdocs.nav.Page` instance
    : __config:__ global configuration object

    Returns:
    : The raw source for a page as unicode string. If `None` is returned, the
      default loading from a file will be performed.

##### on_page_markdown

:   The `page_markdown` event is called after the page's markdown is read and
    can be used to alter the Markdown source text. The meta-data has been 
    stripped off and is available as `page.meta` at this point.
```

Reply:
> what's bugging me is, that 'page', imo, points to the Page class which is the agent here, but not the object that is read.

Fair point. Which reminds me of my original objection. We seem to be having trouble fitting this event in with the other events, which suggests that maybe it doesn't belong as-is. Each of us are suggesting slightly different takes on the documentation. In each case, we each are compromising on a different aspect of the implementation. Neither of us are satisfied as there is a shortfall either way. The way I see it, if we can't write documentation that fits clearly, then we don't have a solution that makes sense. And that's why I'm making such a big deal out of this. Normally I wouldn't be so picky about a single sentence in documentation.

So, let me break down how I see this new event [^1] ... 

This event is called when populating an instance of the `Page` class. And this event would be called once for each "page". It even has the (unpopulated) `Page` instance passed in as a parameter. Therefore, it is a "page" event and "page" needs to be in the name. 

Its actual function (by default) is to read the contents of a file. Therefore it is a "read" event and "read" should be in the name.

I had wanted "markdown" in the name before because I hadn't clarified in my mind the distinction  between "reading" and "loading", but now that we are using "read", my reasons for including  "markdown" aren't that compelling. We don't want to confuse this event and the event which immediately follows: `on_page_markdown`. The later  is where modification of the Markdown text itself should happen. Whereas this event should simply be returning the contents of a file (whether a real file or some pseudo file).  It's not really the concern of this event if the contents of the file happen to not actually be Markdown[^2]. Therefore, I think we can drop the use of "markdown" from the name. 

However, to make it clear that the event does not "read" the `Page` instance, but something else, I agree that including a third word after "read" helps[^3]. I considered  `on_page_read_file`, but that feels slightly contradictory as the whole purpose of this event is to allow reading from something other than a file. What about `on_page_read_src`? This event fires when a `Page` instance 'reads' its 'source', regardless of what that source may happen to be.


[^1]: In fact, this comment went through multiple drafts as I have clarified my take on this. Thats often how I develop concepts and processes--by typing out an explanation. And that explains why many of my comments are so lengthy. 

[^2]: For example, a file could contain Markdown mixed with template variables. Those variables should only be swapped out for their values in the `on_page_markdown` event, after the meta-data is available. In that case, this "read" event would not actually return valid Markdown, but the raw file source.

[^3]: That's also why I suppose you suggested `page_read` rather than `read_page`. I agree, that is an important distinction to make, yet one I hadn't considered previously.